### ISSUE TITLE
Can't able to delete dependencies properly as build method being called after onExit

### ISSUE THREAD

**Issue by tejHackerDEV** [SchabanBo/qlevar_router#75]
I am using GetX to register dependencies. I moved from GetX Navigation to Qlevar for stable Navigator 2.0 support. So to achieve Bindings concept present in GetX for dynamic coupling & decoupling I was using the Middleware concept of Qlevar where on onInit I was using Get.put to register & on onExit I was using Get.delete to delete the dependencies. Everything was working fine but I notice something while working with Nested Navigation ie., build method of a Widget is being called after onExit function of Qlevar so as I am deleting the GetXController in onExit callback of qlevar, I am getting an usual red screen error of Flutter when closing the page saying controller has already removed & error stack. So I am thinking to have an extra callback named onExited where we can delete the dependencies as the page gone out of scope & at no point build method will never get called as it has gone completely out of the scope, which results in no errors and also its a nice to have feature In my opinion, as it can help in do lot more not only deleting dependencies like showing an alert asking user rating or something like that.

**Comment by tejHackerDEV** [comment]
It has nothing to

**Comment by tejHackerDEV** [comment]
Sorry it has nothing to do with the package, it was something that is happening because of using setState on router listener for hiding the appBar & bottomBar but again is there any way to hide the parent appBar & bottomBar when navigating to child page. Because I have a parent with a scaffold containing appBar & bottomBar. For body I passed the router that I got from Qlevar, so when navigating between Childs the scaffold body is changed but the scaffold appBar & bottomBars remains same. So to avoid it I had custom implementation where I am toggling the boolean values via setState to hide appBar & bottomBar but is there any way to make child occupy the whole page by covering the parent appBar & bottomBars
